doc/man3/OSSL_LIB_CTX.pod

```diff
@@ -38,10 +39,17 @@ from a configuration.
 OSSL_LIB_CTX_free() frees the given I<ctx>, unless it happens to be the
 default OpenSSL library context.
 
+OSSL_LIB_CTX_get0_global_default() returns a concrete (non NULL) reference to
+the global default library context.
```

Comment:
That isn't really correct - in that if you are in the FIPS module then it returns NULL. 
Explaining that different context for the lib_ctx should be in this section.

Reply:
Err..no. This is *end* user documentation. End users are *never* in the FIPS module. Calling it from our own code inside the FIPS module would be considered an error (there is no global default there).

Reply:
Err..no. Your are documenting a function that exists in multiple contexts. And we have users that will indeed modify the FIPS provider. Our documentation should not send them astray ... which this does by omitting that context.

Reply:
Really, really - no. Users that modify our own code do so at their own risk. There are many, many, many many, functions that behave differently, or incorrectly, or unexpectedly in the FIPS module compared to normal code or simply don't exist there at all. This is not something that belongs in the end user documentation. We haven't done that for any of these other cases (and nor should we).

Reply:
I could not even imagine anyone calling that function within the FIPS_MODULE code no matter how much patched.

Reply:
Maybe it should just be inside #ifndef FIPS_MODULE as a whole?

Reply:
> Maybe it should just be inside #ifndef FIPS_MODULE as a whole?

Yes - that makes sense. Done.

Reply:
Now it makes more sense - if the function exists and can be called it should be documented where its behaviour is different as there are users who will use it - if the function doesn't exist in the FIPS_MODULE context then not having it is better than having a function that would have effectively non-documented behaviour. And that applies to the other similar functions as well.

Reply:
It does raise the question as to why OSSL_LIB_CTX_set0_default exists in the FIPS_MODULE context ... shouldn't it also not exist there for the same sorts of reasons. Or be documented as to why.

Reply:
I've made the same change to that function - so it no longer exists in the FIPS module.